src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -207,10 +264,15 @@ private void encodeChainedFeatures(String attributeName, List<Feature> chainedFe
         jsonWriter.key(attributeName);
         jsonWriter.array();
         for (Feature feature : chainedFeatures) {
-            // encode each chained feature
-            jsonWriter.object();
-            encodeProperties(null, feature.getType(), feature.getProperties());
-            jsonWriter.endObject();
+            // if it's GeoJSON compatible, encode as a full blown GeoJSON feature (must have a
+            // default geometry)
+            if (feature.getType().getGeometryDescriptor() != null) {
```

Comment:
Question, should we also not do the ``isFullFeature`` test here? Maybe not, since only real features may appear as a list.

Reply:
Good idea, does not hurt to make an extra check.